src/nvim/window.c

```diff
@@ -6528,6 +6533,16 @@ static win_T *get_snapshot_focus(int idx)
 int switch_win(win_T **save_curwin, tabpage_T **save_curtab, win_T *win, tabpage_T *tp, int no_display)
 {
   block_autocmds();
+  return switch_win_noblock(save_curwin, save_curtab, win, tp, no_display);
+}
+
+// As switch_win() but without blocking autocommands.
+int switch_win_noblock(win_T **save_curwin,
```

Comment:
we should also use the "noblock" variants in `nvim_win_set_buf`

Reply:
done! and add a testcase in test/functional/autocmd/window_spec.lua .